### PR TITLE
MOSIP-32454: Merging java21 changes

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: registration/configure.sh [hunk 1]
```diff
@@ -100,19 +100,19 @@ echo "FOR /F \"tokens=* delims=\" %%x in (.UNKNOWN_JARS) DO DEL /Q lib\%%x" >> "
...
 echo ") else (" >> "${work_dir}"/registration-client/target/run.bat
 echo "echo Starting Registration Client" >> "${work_dir}"/registration-client/target/run.bat
-echo "start jre\bin\javaw -Xmx2048m -Xms2048m -Dfile.encoding=UTF-8 -cp lib/*;/* io.mosip.registration.controller.Initialization > startup.log 2>&1" >> "${work_dir}"/registration-client/target/run.bat
+echo "start jre\bin\javaw -Xmx2048m -Xms2048m -Dfile.encoding=UTF-8 -cp lib/*;/* --add-modules=javafx.controls,javafx.fxml --add-exports javafx.graphics/com.sun.javafx.application=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED io.mosip.registration.controller.Initialization > startup.log 2>&1" >> "${work_dir}"/registration-client/target/run.bat
 echo ")" >> "${work_dir}"/registration-client/target/run.bat
 
-cp "${work_dir}"/registration-client/target/run.bat "${work_dir}"/registration-client/target/lib/114to1201_run.bat
+cp "${work_dir}"/registration-client/target/run.bat "${work_dir}"/registration-client/target/lib/1201to121_run.bat
```
Comment (ase-101) [inline registration/configure.sh:109]: do we need any code change for this batch file to be downloaded during the upgrade in the previous version of reg-client or the new version of reg-client?

File: registration/registration-services/pom.xml [hunk 2]
```diff
@@ -474,7 +481,7 @@
 		<dependency>
```
Comment (ase-101) [inline registration/registration-services/pom.xml:481]: We should remove stub-impl dependency from the pom, instead, this should be dynamically loaded when building the reg-client zip in the docker.

File: registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java [hunk 3]
```diff
@@ -509,7 +510,7 @@ private Map<String, Object> getIdaAuthResponse(Map<String, Object> idaRequestMap
 
 			LOGGER.info(LOG_REG_USER_ONBOARD, APPLICATION_NAME, APPLICATION_ID, "Getting Symmetric Key.....");
 			// Symmetric key alias session key
-			KeyGenerator keyGenerator = KeyGeneratorUtils.getKeyGenerator("AES", 256);
+			KeyGenerator keyGenerator = KeyGeneratorUtils.getKeyGenerator("AES", 256, new SecureRandom());
```
Comment (ase-101) [inline registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java:513]: Can we initialize the secureRandom once and reuse it? @mahammedtaheer  what do you suggest?

File: registration/registration-services/src/main/resources/initial.sql [hunk 4]
```diff
@@ -252,7 +252,7 @@ INSERT INTO "REG"."GLOBAL_PARAM" VALUES ('mosip.kernel.applicant.type.age.limit'
 
 INSERT INTO "REG"."GLOBAL_PARAM" VALUES ('mosip.kernel.jsonvalidator.file-storage-uri','mosip.kernel.jsonvalidator.file-storage-uri','LOCAL','CONFIGURATION','eng',true,'SYSTEM',current timestamp, 'SYSTEM',current timestamp, false, current timestamp);
 
-INSERT INTO "REG"."GLOBAL_PARAM" VALUES ('mosip.kernel.crypto.symmetric-algorithm-name','mosip.kernel.crypto.symmetric-algorithm-name','AES/GCM/PKCS5Padding','CONFIGURATION','eng',true,'SYSTEM',current timestamp, 'SYSTEM',current timestamp, false, current timestamp);
+INSERT INTO "REG"."GLOBAL_PARAM" VALUES ('mosip.kernel.crypto.symmetric-algorithm-name','mosip.kernel.crypto.symmetric-algorithm-name','AES/GCM/NoPadding','CONFIGURATION','eng',true,'SYSTEM',current timestamp, 'SYSTEM',current timestamp, false, current timestamp);
```
Comment (ase-101) [inline registration/registration-services/src/main/resources/initial.sql:255]: I don't see any upgrade DB script for this change, is it not required?

**Review by ase-101 — COMMENTED** [review]
Please check comments

File: registration/configure.sh (hunk 1 above)
Comment (HimajaDhanyamraju2) [reply registration/configure.sh:114]: Cross-checked the code. There are no references of .bat file in the code.

File: registration/registration-services/pom.xml (hunk 2 above)
Comment (HimajaDhanyamraju2) [reply registration/registration-services/pom.xml:481]: Removed the dependency from here. Changes done to take the jar from artifactory-ref-impl

File: registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java (hunk 3 above)
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/java/io/mosip/registration/service/operator/impl/UserOnboardServiceImpl.java]: Done.

File: registration/registration-services/src/main/resources/initial.sql (hunk 4 above)
Comment (HimajaDhanyamraju2) [reply registration/registration-services/src/main/resources/initial.sql:255]: Added upgrade scripts.

File: registration/configure.sh (hunk 1 above)
Comment (ase-101) [reply registration/configure.sh:114]: This should be part of upgrade documentation or in the readme file, for the user to copy the file from lib directory to reg-client parent directory.

**Review by ase-101 — APPROVED** [review]

**Pull request merged**